File: store/item/android/src/main/java/com/jakewharton/sdksearch/store/item/DbModule.java
package com.jakewharton.sdksearch.store.item;

import android.content.Context;
import android.support.annotation.Nullable;
import com.squareup.sqldelight.android.SqlDelight;
import dagger.Binds;
import dagger.Module;
import dagger.Provides;

@Module //
abstract class DbModule {
  @Provides static QueryWrapper queryWrapper(Context context, @Nullable String filename) {
    return new QueryWrapper(SqlDelight.create(QueryWrapper.Companion, context, filename));
  }
  
  @Provides static ItemQueries itemQueries(QueryWrapper queryWrapper) {
    return queryWrapper.getItemQueries();
  }

  @Binds abstract ItemStore itemStore(SqlItemStore impl);
}

File: store/item/android-sql/src/main/java/com/jakewharton/sdksearch/store/item/DbModule.kt
package com.jakewharton.sdksearch.store.item

import android.content.Context
import com.squareup.sqldelight.android.AndroidSqliteDriver
import dagger.Binds
import dagger.Module
import dagger.Provides

@Module(includes = [DbModule.Providers::class])
internal abstract class DbModule {
  @Binds abstract fun itemStore(impl: SqlItemStore): ItemStore

  @Module
  internal object Providers {
    @JvmStatic
    @Provides
    fun itemDatabase(context: Context, filename: String?): ItemDatabase {
      return ItemDatabase(AndroidSqliteDriver(ItemDatabase.Schema, context, filename))
    }

    @JvmStatic
    @Provides
    fun itemQueries(database: ItemDatabase): ItemQueries {
      return database.itemQueries
    }
  }
}
